bindings/go/src/fdb/database.go

```diff
@@ -157,19 +159,47 @@ func retryable(wrapped func() (interface{}, error), onError func(Error) FutureNi
 // See the Transactor interface for an example of using Transact with
 // Transaction and Database objects.
 func (d Database) Transact(f func(Transaction) (interface{}, error)) (interface{}, error) {
+	return d.transact(context.Background(), f)
+}
+
+// TransactWithContext - same as Transact but uses context for setting timeout
+func (d Database) TransactWithContext(ctx context.Context, f func(Transaction) (interface{}, error)) (interface{}, error) {
+	return d.transact(ctx, f)
+}
+
+func (d Database) transact(ctx context.Context, f func(Transaction) (interface{}, error)) (interface{}, error) {
 	tr, e := d.CreateTransaction()
 	// Any error here is non-retryable
 	if e != nil {
 		return nil, e
 	}
 
+	if ctx != nil {
+		if deadline, ok := ctx.Deadline(); ok {
```

Comment:
For `context.Background()` which is supposed to have no deadline, does this mean it will end immediately?